lib/model.js

```diff
@@ -340,7 +344,29 @@ ModelBaseClass.prototype._initProperties = function(data, options) {
       }
     }
   }
-  this.trigger('initialize');
+  this.__trigger('initialize');
+};
+
+ModelBaseClass.prototype.__trigger = function trigger(actionName, work, data, callback) {
```

Comment:
Since `__trigger` is used only for `initialize` and always called as `trigger('initialize')`, we can remove the last three arguments and simplify the code below a lot.

It makes me wonder why cannot we simply change L324 to directly call `this.afterInitialize`? I think it has the same effect as the current solution.

```diff
- this.trigger('initialize');
+ if (this.afterInitialize)
+   this.afterInitialize();
+ else if (this.postInitialize) {
+  this.postInitialize();
```